core/src/main/java/hudson/diagnosis/OldDataMonitor.java

```diff
@@ -207,7 +205,7 @@ public static void report(Saveable obj, Collection<Throwable> errors) {
         }
         if (buf.length() == 0) return;
         Jenkins j = Jenkins.getInstanceOrNull();
-        if (j == null) {
+        if (j == null) { // TODO confirm this can never be true
```

Comment:
Probably it _was_ true, which is why this code block was written: startup failed, Jenkins shut down, and now we are trying to report stuff. Maybe things were refactored since then to make this impossible, or maybe the author misdiagnosed a `NullPointerException`, etc.